commands/box/util/root.go

```diff
@@ -0,0 +1,238 @@
+// Copyright (c) 2018 ContentBox Authors.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package utilcmd
+
+import (
+	"encoding/base64"
+	"encoding/hex"
+	"fmt"
+	"math"
+	"strconv"
+
+	root "github.com/BOXFoundation/boxd/commands/box/root"
+	"github.com/BOXFoundation/boxd/core/txlogic"
+	"github.com/BOXFoundation/boxd/core/types"
+	"github.com/BOXFoundation/boxd/crypto"
+	"github.com/btcsuite/btcutil/base58"
+	"github.com/spf13/cobra"
+)
+
+// rootCmd represents the base command when called without any subcommands
+// rootCmd represents the base command when called without any subcommands
+var rootCmd = &cobra.Command{
+	Use:   "util [command]",
+	Short: "some useful gadgets",
+	// Uncomment the following line if your bare application
+	// has an action associated with it:
+	//	Run: func(cmd *cobra.Command, args []string) { },
+}
+
+// Init adds the sub command to the root command.
+func init() {
+	root.RootCmd.AddCommand(rootCmd)
+	rootCmd.AddCommand(
+		&cobra.Command{
+			Use:   "encode [base58/base64][data]",
+			Short: "convert data from hex format to base58/base64 format",
+			Run:   encode,
+		},
+		&cobra.Command{
+			Use:   "decode [base58/base64][data]",
+			Short: "convert data from base58/base64 format to hex format",
+			Run:   decode,
+		},
+		&cobra.Command{
+			Use:   "convertaddress [address]",
+			Short: "convert address to a special format",
+			Run:   convertAddress,
+		},
+		&cobra.Command{
+			Use:   "makesplitaddress [tx_hash] [preOutPoint_index] [(addr1, weight1), (addr2, weight2), (addr3, weight3), ...]",
+			Short: "creat split address",
+			Run:   makeSplitAddress,
+		},
+		&cobra.Command{
+			Use:   "makeContractAddress [fromaddress] [nonce]",
+			Short: "creat contract address",
+			Run:   makeContractAddress,
+		},
+		&cobra.Command{
+			Use:   "makep2pKHAddr [pubkey]",
+			Short: "creat p2pKH address",
+			Run:   makeP2PKHAddress,
+		},
+	)
+}
+
+func encode(cmd *cobra.Command, args []string) {
```

Comment:
⇒  ./box util encode  1
panic: runtime error: index out of range [1] with length 1

goroutine 1 [running]:
github.com/BOXFoundation/boxd/commands/box/util.encode(0xc00026ca00, 0xc00024a2b0, 0x1, 0x1)
	/Users/jerrypeen/go/src/github.com/BOXFoundation/boxd/commands/box/util/root.go:74 +0x3a8